src/utils/astToString.ts

```diff
@@ -1,4 +1,84 @@
-import { generate } from 'astring'
-import { Node } from 'estree'
+import * as astring from 'astring'
 
-export const astToString = (node: Node): string => generate(node)
+import { Node } from '../types'
+
+function reindent(state: any, text: any, indent: any, lineEnd: any) {
+  /*
+    Writes into `state` the `text` string reindented with the provided `indent`.
+    */
+  const lines = text.split('\n')
+  const end = lines.length - 1
+  state.write(lines[0].trim())
+  if (end > 0) {
+    state.write(lineEnd)
+    for (let i = 1; i < end; i++) {
+      state.write(indent + lines[i].trim() + lineEnd)
+    }
+    state.write(indent + lines[end].trim())
+  }
+}
+
+function formatComments(state: any, comments: any, indent: any, lineEnd: any) {
+  /*
+    Writes into `state` the provided list of `comments`, with the given `indent` and `lineEnd` strings.
+    Line comments will end with `"\n"` regardless of the value of `lineEnd`.
+    Expects to start on a new unindented line.
+    */
+  const { length } = comments
+  for (let i = 0; i < length; i++) {
+    const comment = comments[i]
+    state.write(indent)
+    if (comment.type[0] === 'L') {
+      // Line comment
+      state.write('// ' + comment.value.trim() + '\n', comment)
+    } else {
+      // Block comment
+      state.write('/*')
+      reindent(state, comment.value, indent, lineEnd)
+      state.write('*/' + lineEnd)
+    }
+  }
+}
```

Comment:
Minor nit: documentation should use double stars:

```ts
/**
 * Something
 * Something
 */
function f() { /* ... */ }
```

And written above (outside) the function declaration.